graphbolt/src/cnumpy.cc

```diff
@@ -121,78 +119,77 @@ torch::Tensor OnDiskNpyArray::IndexSelectIOUring(torch::Tensor index) {
       0, num_index, group_size_, [&](int64_t begin, int64_t end) {
         auto thread_id = torch::get_thread_num();
         if (!error_flag.load()) {
-          for (int64_t i = begin; i < end; i++) {
-            int64_t group_id = i - begin;
-            int64_t feature_id = index_data[i];  // Feature id.
-            if (feature_id >= feature_dim_[0]) {
-              error_flag.store(true);
-              break;
-            }
-            int64_t offset = feature_id * feature_size_ + prefix_len_;
-            int64_t aligned_offset = offset & (long)~(kDiskAlignmentSize - 1);
-            residual[thread_id * group_size_ + group_id] =
-                offset - aligned_offset;
-
-            int64_t read_size;
-            if (residual[thread_id * group_size_ + group_id] + feature_size_ >
-                kDiskAlignmentSize) {
-              read_size = aligned_length + kDiskAlignmentSize;
-            } else {
-              read_size = aligned_length;
+          for (int64_t batch_start = begin; batch_start < end;
+               batch_start += group_size_) {
```

Comment:
this for loop looks not straightfoward to me. You're submitting the queue when it's full? Is it possible to implement like below?
```
for(int64_t i = begin; i < end; ++i){
  //add read request to queue.
  // submit if queue is full: ==group_size_?
 // handle read data.
}
```